src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/DistanceSlopeCalculator.java

```diff
@@ -0,0 +1,61 @@
+/* (c) 2025 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wps.longitudinal;
+
+import org.geotools.api.referencing.crs.CoordinateReferenceSystem;
+import org.geotools.api.referencing.crs.GeographicCRS;
+import org.geotools.api.referencing.operation.TransformException;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.referencing.GeodeticCalculator;
+import org.locationtech.jts.geom.Point;
+
+class DistanceSlopeCalculator {
+
+    GeodeticCalculator gc;
+    Point current;
+    Point previous;
+    CoordinateReferenceSystem projection;
+    double distance;
+    double slope;
+
+    public DistanceSlopeCalculator(CoordinateReferenceSystem projection) {
```

Comment:
projection is a constructor dependency, we may make the field final if no further change is expected.